### PR TITLE
Allow the user to specify an arbitrary number of tabs

### LINKED ISSUE #2512: More than five tabs
Please can we have more than five tabs - either by squashing them together a bit more or by allowing to scroll the tabs bar left/right to see the ones which are out of view?

Home, Mentions, Local, Federated, Direct Messages is already five tabs before adding ones for lists or hashtags.

### PR DESCRIPTION
**Pull request by nikclayton**
Fixes #2512

### CODE REVIEW

**Comment by nikclayton** [comment]
+20,000 !!! -- don't panic, that's changes to the lint baseline file (this PR removes some existing lint errors as a nice side effect).

**Comment by nikclayton** [comment]
This changes the initial tab presentation. Android behaviour seems to be to size the tabs appropriately and leave space, like this.

image

I think that's fine -- it's a clear indiciation to the user "Hey, you can put more stuff here".

The tabs to the right may or may not be partially visible, depending on the user's other settings.

For example, with 6 tabs configured:

image

the extra tab is peeking in from the right. But if the user has turned off the top nav it looks like this:

image

and it's not apparent there are extra tabs.

However, those extra tabs become visible when the user starts swiping through the tabs or tapping on the icons. For example, after tapping on the DM icon the icons scroll, and it's immediately obvious there is more content.

That looks like this:

image

and this (if top nav is turned off):

image

To forestall the obvious comments:

1. We can't remove the avatar icon, that's the accessible way for the user to open the navigation menu
2. We can't remove the vertical-ellipsis menu, that's the accessible way for the user to refresh and access other functionality that is normally gesture-based.

**Comment by nikclayton** [comment]
Additional enhancements worth considering, but I'm not going to do as part of this PR:

- Add a fixed tab on the right that, when clicked, opens the tab preferences UI, so the user can adjust this with one click instead of three (tap avatar to open nav menu, tap "Account preferences", tap "Tabs")
- Allow the user to specify custom tab titles, and show those titles

Existing bugs that this PR does not fix, and will not fix:

- The user can add multiple tabs of the same type (e.g., multiple DM tabs). I explain how to fix that in https://github.com/tuskyapp/Tusky/issues/3331 (you can think of this PR as being part of the set of work necessary to implement the vision set out in that issue).

File: app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt [hunk 1]
```diff
@@ -421,6 +414,5 @@ class TabPreferenceActivity : BaseActivity(), Injectable, ItemInteractionListene
 
     companion object {
         private const val MIN_TAB_COUNT = 2
```
Comment (Lakoja) [inline app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt:416]: Better have 1 here?

(2 seems arbitrary, 1 is a "guard against total misconfiguration")

**Comment by Lakoja** [comment]
> +20,000 !!! -- don't panic, that's changes to the lint baseline file (this PR removes some existing lint errors as a nice side effect).

Obsidian? A JS file?

**Comment by nikclayton** [comment]
> > +20,000 !!! -- don't panic, that's changes to the lint baseline file (this PR removes some existing lint errors as a nice side effect).
> 
> Obsidian? A JS file?

Oh, WTF is that doing there. One sec...

**Pull request closed without merging**